core/src/main/scala/kafka/tools/StorageTool.scala

```diff
@@ -102,6 +111,23 @@ object StorageTool extends Logging {
       setClusterId(clusterId).
       setNodeId(config.nodeId).
       build()
+    val standaloneMode = namespace.getBoolean("standalone")
+    var advertisedListenerEndpoints: collection.Seq[kafka.cluster.EndPoint] = List()
+
+    val controllersQuorumVoters = namespace.getString("controller_quorum_voters")
+    if(standaloneMode && controllersQuorumVoters != null) {
+      throw new TerseFailure("Both --standalone and --controller-quorum-voters were set. Only one of the two flags can be set.")
+    }
+
+    if (standaloneMode) {
+      advertisedListenerEndpoints = config.effectiveAdvertisedBrokerListeners
```

Comment:
The bootstrapping checkpoint should only be generated on controllers (`config.processRoles.contains(ProcessRole.ControllerRole)`). If the node is a broker only then the bootstrapping checkpoint should not get generated.